File: cindyberg/QLJava/src/org/uva/sea/ql/ast/type/StringType.java
package org.uva.sea.ql.ast.type;

import org.uva.sea.ql.ast.IVisitor;
import org.uva.sea.ql.ast.TypeVisitor;


public class StringType extends Type{

	public void accept(IVisitor visitor) {
		visitor.visit(this);		
	}
	
	@Override
	public String toString(){
		return "string";
	}
	
	@Override
	public boolean isCompatibleWith(Type t) {
		return t.isCompatibleWithString();
	}
	
	@Override
	public boolean isCompatibleWithString(){
		return true;
	}

	@Override
	public <T> T accept(TypeVisitor<T> visitor) {
		return visitor.visit(this);
	}
}

File: rperz/QSLib/QSLib/AST/Types/StringType.cs
using System.Windows;
using System.Windows.Controls;
using QSLib.GUI;
using QSLib.AST.Types;
using QSLib.Visitors;

namespace QSLib.AST.Types
{
    public class StringType : QSType
    {
        public override string Name
        {
            get { return "String"; }
        }

        public override bool IsCompatible(QSType other)
        {
            if (other == null)
                return false;
            return other.IsString();
        }

        public override bool IsString()
        {
            return true;
        }

        public override string ToString()
        {
            return "string";
        }


        public override object Accept<T>(ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }


        public override object GetUndefined()
        {
            return "";
        }
    }
}
